lib/server/routes/reports.js

```diff
@@ -72,17 +151,108 @@ ReportsRouter.prototype.createExchangeReport = function(req, res, next) {
   });
 };
 
+/**
+ * Creates an exchange report
+ * @param {http.IncomingMessage} req
+ * @param {http.ServerResponse} res
+ * @param {Function} next
+ */
+ReportsRouter.prototype.createExchangeReport = function(req, res, next) {
+  const token = req.body.token;
+
+  /* jshint maxstatements: 30, maxcomplexity: 15 */
+  this.storage.models.StorageEvent.findOne({token: token}, (err, event) => {
+    if (err) {
+      return next(new errors.InternalError(err.message));
+    }
+
+    if (!event) {
+      return next(new errors.NotFoundError('Storage event not found'));
+    }
+
+    // TODO: Check that event isn't closed for reports
+
+    const report = {
+      exchangeStart: req.body.exchangeStart,
+      exchangeEnd: req.body.exchangeEnd,
+      exchangeResultCode: req.body.exchangeResultCode,
+      exchangeResultMessage: req.body.exchangeResultMessage
+    };
+
+    if (!this.validateExchangeReport(report)) {
+      return next(new errors.BadRequestError('Invalid exchange report'));
+    }
+
+    const isClientReport = req.user ?
+          (event.client === req.user.id) : req.farmerNodeID ?
+          (event.client === req.farmerNodeID) : false;
+    const isFarmerReport = req.farmerNodeID ?
+          (event.farmer === req.farmerNodeID) : false;
+
+    if (!isClientReport && !isFarmerReport) {
+      return next(new errors.NotAuthorizedError('Not authorized to report'));
+    }
+
+    let modified = false;
+
+    const hasClientReport = event.clientReport ?
+          !!event.clientReport.exchangeResultCode : false;
+    const hasFarmerReport = event.farmerReport ?
+          !!event.farmerReport.exchangeResultCode : false;
+
+    if (isClientReport && !hasClientReport) {
+      modified = true;
+      event.clientReport = report;
+      if (Number(req.body.exchangeResultCode) === 1000) {
+        event.success = true;
+        this.updateReputation(event.farmer, points.TRANSFER_SUCCESS);
+      } else {
+        this.updateReputation(event.farmer, points.TRANSFER_FAILURE);
```

Comment:
In case of mirror creation I will send a ClientReport. By claiming all mirror transfers as failed I can reduce the reputation of all my neighbors. Better chance for myself.

Reply:
Agree, we can take a look at this later. We would likely need to do something similar as we have with users.